### PR TITLE
Rework fontsize, better defaults

### LINKED ISSUE #1403: Tab font size too small
### Jerboa Version

0.0.60

### Android Version + Phone

Android 14, Google Pixel 7

### Describe The Bug

Font size of about, posts and comments tabs in the profile section is considerably smaller than in previous versions. Almost can't see it. 

Applies to tabs across the board, in messages too.

Notified you of this on Lemmy, opening the issue as per your request. Thanks for all the great work. 

### To Reproduce

Just run v0.0.60 and pay attention to tabs font size. 

### In the case of a crash or when relevant include the logs

_No response_

### PR DESCRIPTION
**Pull request by MV-GH**
|                | Compose default in SP | Jerboa default in Px | Jerboa default in SP now |
|----------------|-----------------------|----------------------|----------------------|
| headlineLarge  | 32                    | 16                   | 22* |
| headlineMedium | 28                    | 13                   | 16* |
| headlineSmall  | 24                    | 9                    | 14* |
| titleLarge     | 22                    | 16                   |  22                    | 
| titleMedium    | 16                    | 13                   | 16                    |
| titleSmall     | 14                    | 9                    | 14                    |
| bodyLarge      | 16                    | 16                   | 16                    
| bodyMedium     | 14                    | 13                   | 14                    |
| bodySmall      | 12                    | 9                    | 12                    |
| labelLarge     | 14                    | 16                   | 14                    |
| labelMedium    | 12                    | 13                   | 12                    | 
| labelSmall     | 11                    | 9                    | 11                    |
| displayLarge   | 57                    | x                    | x |
| displayMedium  | 45                    | x                    | x |
| displaySmall   | 36                    | x                    | x |

I made it scale with the ratio of the bodyLarge, meaning that if you set font size to 14 bodyLarge becomes 14.
And the other styles resize at the same ratio (14/16 = 87.5% size)

Fixes #1403 

I am not done yet. But I recommend checking out if you see anything standing out. I made it so that most fields stay the same.

### CODE REVIEW

File: app/src/main/java/com/jerboa/ui/theme/Type.kt [hunk 1]
```diff
@@ -1,76 +1,61 @@
...
-const val FONT_MEDIUM_ADD = -3
-const val FONT_LARGE_ADD = 0
-const val LINE_HEIGHT_MULTIPLIER = 1.5
+const val COMPOSE_BODY_LARGE_DEFAULT = 16
+val BASE_TYPOGRAPHY = Typography()
 
-// Set of Material typography styles to start with
-// Example from here: https://egeniq.com/blog/dynamic-font-sizes-with-jetpack-compose/
 fun generateTypography(baseFontSize: Int): Typography {
+    val customRatio = baseFontSize.toFloat() / COMPOSE_BODY_LARGE_DEFAULT
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/theme/Type.kt:9]: This approach does seem much simpler, thx.

**Comment by MV-GH** [comment]
Somethings I don't get

How it looked the old way:

image

this is twice titleMedium somehow `new` has more weight??

but using the new system

titleMedium is bigger but (even when only 14 vs original 13)
Both have the same font weight

image

**Comment by MV-GH** [comment]
Made it look like this

image


The documented changes are:
- HeaderTitle
- Tabs
- PostAndCommunityContextHeader
- SimpleTopBar (Indifferent to this one was 16 now is 22, kept compose default)

Everything else should be the exact same as before

Opinions?

File: app/src/main/java/com/jerboa/ui/components/common/AppBars.kt [hunk 2]
```diff
@@ -811,3 +813,35 @@ fun ActionTopBar(
         },
     )
 }
+
+@Composable
+fun DualHeaderTitle(
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/components/common/AppBars.kt:818]: Thx for extracting this.

**Review by dessalines — APPROVED** [review]
Just tested this, and it fixed the primary issue, and sizing works much better now. Thx!

**Pull request merged**